[Automation] native-image run fails for com.google.cloud:grpc-gcp:1.11.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26938218184
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
amcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.cloud/grpc-gcp/1.10.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.protobuf/protobuf-java/2.5.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.protobuf/protobuf-java-util/3.23.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.code.findbugs/jsr305/3.0.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.code.gson/gson/2.2.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.errorprone/error_prone_annotations/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-api/1.27.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/guava/33.6.0-jre,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/failureaccess/1.0.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jspecify/jspecify/1.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.j2objc/j2objc-annotations/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-core/1.69.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.android/annotations/4.1.1.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.codehaus.mojo/animal-sniffer-annotations/1.23,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.perfmark/perfmark-api/0.25.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-context/1.57.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-netty-shaded/1.68.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-protobuf/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api.grpc/proto-google-common-protos/2.41.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-stub/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opencensus/opencensus-api/0.31.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-api/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-context/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-common/1.56.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api/api-common/2.37.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.auto.value/auto-value-annotations/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=38a7ee22-9dde-77de-a452-e08509e8167a \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'com.google.cloud.grpc-gcp_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (9.3s @ 0.18GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (69.0s @ 1.62GB)
   14,063 types,  20,156 fields, and  76,369 methods found reachable
    4,990 types,   1,352 fields, and   9,909 methods registered for reflection
       96 types,      92 fields, and     369 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (8.5s @ 1.85GB)
[4/8] Parsing methods...      [***]                                                                      (8.0s @ 1.32GB)
[5/8] Inlining methods...     [***]                                                                      (3.5s @ 1.44GB)
[6/8] Compiling methods...    [******]                                                                  (41.4s @ 1.51GB)
[7/8] Laying out methods...   [***]                                                                      (7.0s @ 1.90GB)
[8/8] Creating image...       [***]                                                                      (7.2s @ 2.22GB)
  21.66MB (31.35%) for code area:    49,698 compilation units
  39.91MB (57.78%) for image heap:  345,804 objects and 162 resources
   7.51MB (10.88%) for other data
  69.08MB in total image size, 64.75MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   8.70MB java.base                                           16.57MB byte[] for general heap data
   3.85MB svm.jar (Native Image)                               6.67MB byte[] for code metadata
   2.40MB grpc-netty-shaded-1.81.0.jar                         4.42MB byte[] for java.lang.String
   1.54MB protobuf-java-4.33.2.jar                             2.27MB java.lang.String
 577.54kB assertj-core-3.22.0.jar                              1.90MB java.lang.Class
 483.10kB java.xml                                             1.46MB com.oracle.svm.core.hub.DynamicHubCompanion
 476.34kB grpc-core-1.81.0.jar                               832.60kB byte[] for reflection metadata
 314.75kB grpc-gcp-1.11.0.jar                                641.38kB heap alignment
 234.41kB java.rmi                                           472.22kB java.util.concurrent.ConcurrentHashMap$Node
 199.12kB junit-jupiter-engine-5.9.2.jar                     413.73kB c.o.svm.core.hub.DynamicHub$ReflectionMetadata
   2.11MB for 48 more packages                                 4.26MB for 3083 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 37 component(s) is embedded in binary (1.73kB). 7 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       14.3s (9.0% of total time) in 1318 GCs | Peak RSS: 3.13GB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'com.google.cloud.grpc-gcp_tests-tests' in 2m 37s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelOptionsRetainConfiguredValuesAndCanBeCopied() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > optionBuildersValidatePoolAndEndpointInputs() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackOptionsConfigureFallbackChannelAndDelegateLifecycle() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackChannelSwitchesToFallbackWhenPrimaryErrorRateThresholdIsReached() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > protoConfigurationSupportsBuildersJsonAndParsing() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelBuilderLoadsApiConfigFromJsonStringAndFile() FAILED

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointChannelRoutesCallsUsingDefaultCallOptionAndContextSelections() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointOptionsCanBeCopiedAndCoreEndpointSelectionTracksSwitches() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Grpc_gcpTest:managedChannelBuilderLoadsApiConfigFromJsonStringAndFile()
    MethodSource [className = 'com_google_cloud.grpc_gcp.Grpc_gcpTest', methodName = 'managedChannelBuilderLoadsApiConfigFromJsonStringAndFile', methodParameterTypes = '']
    => java.lang.IllegalStateException: Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions" missing method "getCtype".
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2140)
       com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:55)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor$ReflectionInvoker.<init>(GeneratedMessage.java:2525)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor.<init>(GeneratedMessage.java:2594)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularEnumFieldAccessor.<init>(GeneratedMessage.java:3065)
       [...]
     Caused by: java.lang.NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions.getCtype()
       java.base@25.0.3/java.lang.Class.checkMethod(DynamicHub.java:1343)
       java.base@25.0.3/java.lang.Class.getMethod(DynamicHub.java:1337)
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2137)
       [...]

Test run finished after 115 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/reports/problems/problems-report.html
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::arrayBaseOffset has been called by com.google.protobuf.UnsafeUtil$MemoryAccessor (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Please consider reporting this to the maintainers of class com.google.protobuf.UnsafeUtil$MemoryAccessor
WARNING: sun.misc.Unsafe::arrayBaseOffset will be removed in a future release
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::loadLibrary has been called by io.grpc.netty.shaded.io.netty.util.internal.NativeLibraryUtil in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Jun 04, 2026 8:08:20 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-2: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:08:20 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-3: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:08:20 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-4: Metric registry is null. Metrics disabled.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 42s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.cloud:grpc-gcp:1.10.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 45s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.0 at 2026-06-04T08:08:20Z with exit code 1 after 165s
❌ ERROR: 'native-image run' for 1.11.0 failed with exit code 1.
FAILED[native-image run][1.11.0][env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26938218184
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
bility-metadata/graalvm-reachability-metadata/metadata/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jspecify/jspecify/1.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.j2objc/j2objc-annotations/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-core/1.69.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.android/annotations/4.1.1.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.codehaus.mojo/animal-sniffer-annotations/1.23,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.perfmark/perfmark-api/0.25.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-context/1.57.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-netty-shaded/1.68.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-protobuf/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api.grpc/proto-google-common-protos/2.41.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-stub/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opencensus/opencensus-api/0.31.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-api/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-context/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-common/1.56.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api/api-common/2.37.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.auto.value/auto-value-annotations/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=38a7ee22-9dde-77de-a452-e08509e8167a \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'com.google.cloud.grpc-gcp_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (8.7s @ 0.28GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (58.2s @ 1.32GiB)
   14,002 types,  20,040 fields, and  74,869 methods found reachable
      677 types,     210 fields, and   3,029 methods registered for reflection
       96 types,      92 fields, and     369 methods registered for JNI access
      162 resource accesses registered with 15.00MiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.8s @ 1.51GiB)
[4/8] Parsing methods...      [***]                                                                     (5.8s @ 1.63GiB)
[5/8] Inlining methods...     [***]                                                                     (3.3s @ 1.74GiB)
[6/8] Compiling methods...    [******]                                                                 (36.3s @ 1.37GiB)
[7/8] Laying out methods...   [***]                                                                     (5.6s @ 1.75GiB)
[8/8] Creating image...       [***]                                                                     (5.4s @ 1.96GiB)
  18.60MiB (30.41%) for code area:    48,203 compilation units
  37.88MiB (61.94%) for image heap:  360,710 objects and 32 resources
   4.68MiB ( 7.65%) for other data
  61.15MiB in total image size, 57.13MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.93MiB o.g.n.runtime/com.oracle.svm.core.code             15.00MiB byte[] for embedded resources
   1.34MiB com.google.protobuf                                 5.69MiB byte[] for code metadata
 948.90KiB java.base/sun.security.ssl                          4.28MiB byte[] for string data
 797.97KiB java.base/java.util                                 2.22MiB java.base/java.lang.String
 473.42KiB java.base/com.sun.crypto.provider                   1.80MiB java.base/java.lang.Class
 441.29KiB io.grpc.internal                                    1.50MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 407.07KiB java.base/jdk.internal.classfile.impl               1.07MiB byte[] for general heap data
 402.29KiB java.base/java.lang                               962.43KiB byte[] for reflection metadata
 378.97KiB io.grpc.netty.shaded.io.netty.buffer              543.36KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 361.65KiB io.grpc.netty.shaded.io.netty.handler.ssl         410.52KiB java.base/java.lang.Object[]
  11.22MiB for 419 more packages                               4.45MiB for 3104 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 32 component(s) is embedded in binary (1.65KiB). 7 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                      10.9s (8.1% of total time) in 1365 GCs | Peak RSS: 2.75GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests (executable, 57.13MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'com.google.cloud.grpc-gcp_tests-tests' in 2m 13s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelOptionsRetainConfiguredValuesAndCanBeCopied() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > optionBuildersValidatePoolAndEndpointInputs() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackOptionsConfigureFallbackChannelAndDelegateLifecycle() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackChannelSwitchesToFallbackWhenPrimaryErrorRateThresholdIsReached() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > protoConfigurationSupportsBuildersJsonAndParsing() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelBuilderLoadsApiConfigFromJsonStringAndFile() FAILED

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointChannelRoutesCallsUsingDefaultCallOptionAndContextSelections() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointOptionsCanBeCopiedAndCoreEndpointSelectionTracksSwitches() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Grpc_gcpTest:managedChannelBuilderLoadsApiConfigFromJsonStringAndFile()
    MethodSource [className = 'com_google_cloud.grpc_gcp.Grpc_gcpTest', methodName = 'managedChannelBuilderLoadsApiConfigFromJsonStringAndFile', methodParameterTypes = '']
    => java.lang.IllegalStateException: Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions" missing method "getCtype".
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2140)
       com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:55)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor$ReflectionInvoker.<init>(GeneratedMessage.java:2525)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor.<init>(GeneratedMessage.java:2594)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularEnumFieldAccessor.<init>(GeneratedMessage.java:3065)
       [...]
     Caused by: java.lang.NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions.getCtype()
       java.base@25.0.3/java.lang.Class.checkMethod(DynamicHub.java:1647)
       java.base@25.0.3/java.lang.Class.getMethod(DynamicHub.java:1641)
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2137)
       [...]

Test run finished after 111 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date
WARNING: sun.misc.Unsafe::arrayBaseOffset has been called by com.google.protobuf.UnsafeUtil$MemoryAccessor (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Please consider reporting this to the maintainers of class com.google.protobuf.UnsafeUtil$MemoryAccessor
WARNING: sun.misc.Unsafe::arrayBaseOffset will be removed in a future release
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::loadLibrary has been called by io.grpc.netty.shaded.io.netty.util.internal.NativeLibraryUtil in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Jun 04, 2026 8:06:09 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-2: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:06:09 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-3: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:06:09 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-4: Metric registry is null. Metrics disabled.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 18s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.cloud:grpc-gcp:1.10.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 20s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.0 at 2026-06-04T08:06:10Z with exit code 1 after 141s
❌ ERROR: 'native-image run' for 1.11.0 failed with exit code 1.
FAILED[native-image run][1.11.0][env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26938218184
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
rk/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jspecify/jspecify/1.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.j2objc/j2objc-annotations/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-core/1.69.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.android/annotations/4.1.1.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.codehaus.mojo/animal-sniffer-annotations/1.23,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.perfmark/perfmark-api/0.25.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-context/1.57.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-netty-shaded/1.68.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-protobuf/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api.grpc/proto-google-common-protos/2.41.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-stub/1.70.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opencensus/opencensus-api/0.31.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-api/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-context/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-common/1.56.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.api/api-common/2.37.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.auto.value/auto-value-annotations/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/javax.annotation/javax.annotation-api/1.2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=dc67f945-36da-abe9-01b5-dec5763d147b \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'com.google.cloud.grpc-gcp_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (8.3s @ 0.28GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (57.1s @ 1.31GiB)
   14,242 types,  20,500 fields, and  76,338 methods found reachable
      695 types,     210 fields, and   3,052 methods registered for reflection
       97 types,      94 fields, and     369 methods registered for JNI access
      164 resource accesses registered with 15.00MiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (7.0s @ 1.52GiB)
[4/8] Parsing methods...      [***]                                                                     (6.1s @ 1.63GiB)
[5/8] Inlining methods...     [***]                                                                     (3.2s @ 1.74GiB)
[6/8] Compiling methods...    [******]                                                                 (36.0s @ 1.40GiB)
[7/8] Laying out methods...   [***]                                                                     (5.8s @ 1.78GiB)
[8/8] Creating image...       [***]                                                                     (5.4s @ 1.98GiB)
  19.07MiB (30.47%) for code area:    49,288 compilation units
  38.56MiB (61.62%) for image heap:  375,390 objects and 34 resources
   4.95MiB ( 7.91%) for other data
  62.58MiB in total image size, 58.44MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.97MiB o.g.n.runtime/com.oracle.svm.core.code             15.00MiB byte[] for embedded resources
   1.34MiB com.google.protobuf                                 5.84MiB byte[] for code metadata
 964.98KiB java.base/sun.security.ssl                          4.41MiB byte[] for string data
 802.42KiB java.base/java.util                                 2.27MiB java.base/java.lang.String
 472.79KiB java.base/com.sun.crypto.provider                   1.83MiB java.base/java.lang.Class
 427.17KiB io.grpc.internal                                    1.52MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 413.08KiB java.base/java.lang                                 1.07MiB byte[] for general heap data
 407.04KiB java.base/jdk.internal.classfile.impl             982.83KiB byte[] for reflection metadata
 378.88KiB io.grpc.netty.shaded.io.netty.buffer              552.73KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 373.77KiB io.grpc.netty.shaded.io.netty.handler.ssl         447.98KiB java.base/java.lang.Object[]
  11.61MiB for 424 more packages                               4.68MiB for 3156 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 32 component(s) is embedded in binary (1.66KiB). 7 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                      10.9s (8.1% of total time) in 1437 GCs | Peak RSS: 2.75GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests (executable, 58.44MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'com.google.cloud.grpc-gcp_tests-tests' in 2m 12s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::arrayBaseOffset has been called by com.google.protobuf.UnsafeUtil$MemoryAccessor (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Please consider reporting this to the maintainers of class com.google.protobuf.UnsafeUtil$MemoryAccessor
WARNING: sun.misc.Unsafe::arrayBaseOffset will be removed in a future release
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::loadLibrary has been called by io.grpc.netty.shaded.io.netty.util.internal.NativeLibraryUtil in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Jun 04, 2026 8:06:05 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-2: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:06:05 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-3: Metric registry is null. Metrics disabled.
Jun 04, 2026 8:06:05 AM com.google.cloud.grpc.GcpManagedChannel initMetrics
INFO: pool-4: Metric registry is null. Metrics disabled.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/native/nativeTestCompile/com.google.cloud.grpc-gcp_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 17s
com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelOptionsRetainConfiguredValuesAndCanBeCopied() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > optionBuildersValidatePoolAndEndpointInputs() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackOptionsConfigureFallbackChannelAndDelegateLifecycle() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > fallbackChannelSwitchesToFallbackWhenPrimaryErrorRateThresholdIsReached() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > protoConfigurationSupportsBuildersJsonAndParsing() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > managedChannelBuilderLoadsApiConfigFromJsonStringAndFile() FAILED

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointChannelRoutesCallsUsingDefaultCallOptionAndContextSelections() SUCCESSFUL

com_google_cloud.grpc_gcp.Grpc_gcpTest > multiEndpointOptionsCanBeCopiedAndCoreEndpointSelectionTracksSwitches() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Grpc_gcpTest:managedChannelBuilderLoadsApiConfigFromJsonStringAndFile()
    MethodSource [className = 'com_google_cloud.grpc_gcp.Grpc_gcpTest', methodName = 'managedChannelBuilderLoadsApiConfigFromJsonStringAndFile', methodParameterTypes = '']
    => java.lang.IllegalStateException: Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions" missing method "getCtype".
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2140)
       com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:55)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor$ReflectionInvoker.<init>(GeneratedMessage.java:2525)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularFieldAccessor.<init>(GeneratedMessage.java:2594)
       com.google.protobuf.GeneratedMessage$FieldAccessorTable$SingularEnumFieldAccessor.<init>(GeneratedMessage.java:3065)
       [...]
     Caused by: java.lang.NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions.getCtype()
       java.base@25.0.3/java.lang.Class.checkMethod(DynamicHub.java:1647)
       java.base@25.0.3/java.lang.Class.getMethod(DynamicHub.java:1641)
       com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:2137)
       [...]

Test run finished after 113 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.cloud/grpc-gcp/1.10.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.cloud:grpc-gcp:1.10.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 19s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.0 at 2026-06-04T08:06:05Z with exit code 1 after 139s
❌ ERROR: 'native-image run' for 1.11.0 failed with exit code 1.
FAILED[native-image run][1.11.0][env GVM_TCK_LV=1.11.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.google.cloud:grpc-gcp:1.10.0]
```

--- a/metadata/com.google.cloud/grpc-gcp/1.11.0/reachability-metadata.json
+++ b/metadata/com.google.cloud/grpc-gcp/1.11.0/reachability-metadata.json
@@ -1,0 +1,1149 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "android.app.Application"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig",
+      "methods": [
+        {
+          "name": "getAffinityKey",
+          "parameterTypes": []
+        },
+        {
+          "name": "getCommandValue",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig",
+      "methods": [
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.AffinityConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig",
+      "methods": [
+        {
+          "name": "getAffinityKey",
+          "parameterTypes": []
+        },
+        {
+          "name": "getCommandValue",
+          "parameterTypes": []
+        },
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.MethodConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig$Builder",
+      "methods": [
+        {
+          "name": "getAffinityKey",
+          "parameterTypes": []
+        },
+        {
+          "name": "getCommandValue",
+          "parameterTypes": []
+        },
+        {
+          "name": "setAffinityKey",
+          "parameterTypes": [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name": "setCommandValue",
+          "parameterTypes": [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.AffinityConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig$Builder",
+      "methods": [
+        {
+          "name": "getAffinityKey",
+          "parameterTypes": []
+        },
+        {
+          "name": "getCommandValue",
+          "parameterTypes": []
+        },
+        {
+          "name": "setAffinityKey",
+          "parameterTypes": [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name": "setCommandValue",
+          "parameterTypes": [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.AffinityConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.AffinityConfig$Command"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.cloud.grpc.proto.ApiConfig",
+      "methods": [
+        {
+          "name": "getChannelPool",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMethodList",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasChannelPool",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.ApiConfig",
+      "methods": [
+        {
+          "name": "getChannelPool",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMethodList",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasChannelPool",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.ApiConfig$Builder",
+      "methods": [
+        {
+          "name": "addMethod",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.MethodConfig"
+          ]
+        },
+        {
+          "name": "getMethodCount",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasChannelPool",
+          "parameterTypes": []
+        },
+        {
+          "name": "setChannelPool",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.ChannelPoolConfig"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.ApiConfig$Builder",
+      "methods": [
+        {
+          "name": "addMethod",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.MethodConfig"
+          ]
+        },
+        {
+          "name": "getMethodCount",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasChannelPool",
+          "parameterTypes": []
+        },
+        {
+          "name": "setChannelPool",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.ChannelPoolConfig"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig",
+      "methods": [
+        {
+          "name": "getIdleTimeout",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxSize",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig",
+      "methods": [
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ChannelPoolConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig",
+      "methods": [
+        {
+          "name": "getIdleTimeout",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxSize",
+          "parameterTypes": []
+        },
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig$Builder",
+      "methods": [
+        {
+          "name": "getMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxSize",
+          "parameterTypes": []
+        },
+        {
+          "name": "setMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": [
+            "int"
+          ]
+        },
+        {
+          "name": "setMaxSize",
+          "parameterTypes": [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ChannelPoolConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.ChannelPoolConfig$Builder",
+      "methods": [
+        {
+          "name": "getIdleTimeout",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": []
+        },
+        {
+          "name": "getMaxSize",
+          "parameterTypes": []
+        },
+        {
+          "name": "setIdleTimeout",
+          "parameterTypes": [
+            "long"
+          ]
+        },
+        {
+          "name": "setMaxConcurrentStreamsLowWatermark",
+          "parameterTypes": [
+            "int"
+          ]
+        },
+        {
+          "name": "setMaxSize",
+          "parameterTypes": [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig",
+      "methods": [
+        {
+          "name": "getAffinity",
+          "parameterTypes": []
+        },
+        {
+          "name": "getNameList",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasAffinity",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig",
+      "methods": [
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.MethodConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig",
+      "methods": [
+        {
+          "name": "getAffinity",
+          "parameterTypes": []
+        },
+        {
+          "name": "getNameList",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasAffinity",
+          "parameterTypes": []
+        },
+        {
+          "name": "newBuilder",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannelBuilder"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig$Builder",
+      "methods": [
+        {
+          "name": "addName",
+          "parameterTypes": [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name": "getNameCount",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasAffinity",
+          "parameterTypes": []
+        },
+        {
+          "name": "setAffinity",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.AffinityConfig"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.MethodConfig"
+      },
+      "type": "com.google.cloud.grpc.proto.MethodConfig$Builder",
+      "methods": [
+        {
+          "name": "addName",
+          "parameterTypes": [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name": "getNameCount",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasAffinity",
+          "parameterTypes": []
+        },
+        {
+          "name": "setAffinity",
+          "parameterTypes": [
+            "com.google.cloud.grpc.proto.AffinityConfig"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FeatureSet"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions",
+      "methods": [
+        {
+          "name": "getDeprecated",
+          "parameterTypes": []
+        },
+        {
+          "name": "getEditionDefaultsList",
+          "parameterTypes": []
+        },
+        {
+          "name": "getTargetsCount",
+          "parameterTypes": []
+        },
+        {
+          "name": "getUninterpretedOptionList",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasCtype",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasDebugRedact",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasDeprecated",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasFeatureSupport",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasFeatures",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasJstype",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasLazy",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasPacked",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasRetention",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasUnverifiedLazy",
+          "parameterTypes": []
+        },
+        {
+          "name": "hasWeak",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$Builder"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$CType"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$EditionDefault"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$FeatureSupport"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$JSType"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$OptionRetention"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$FieldOptions$OptionTargetType"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "type": "com.google.protobuf.DescriptorProtos$UninterpretedOption"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "com.google.protobuf.ExtensionRegistry",
+      "methods": [
+        {
+          "name": "getEmptyRegistry",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.census.InternalCensusStatsAccessor"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.internal.DnsNameResolverProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.grpc.netty.UdsNameResolverProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.grpc.netty.UdsNettyChannelProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.ChannelException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.DefaultFileRegion",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "file"
+        },
+        {
+          "name": "transferred"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.Epoll",
+      "methods": [
+        {
+          "name": "isAvailable",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.EpollDomainSocketChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.EpollEventLoopGroup"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.EpollServerSocketChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.EpollSocketChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.LinuxSocket",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.Native",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.NativeDatagramPacketArray$NativeDatagramPacket",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "count"
+        },
+        {
+          "name": "memoryAddress"
+        },
+        {
+          "name": "recipientAddr"
+        },
+        {
+          "name": "recipientAddrLen"
+        },
+        {
+          "name": "recipientPort"
+        },
+        {
+          "name": "recipientScopeId"
+        },
+        {
+          "name": "segmentSize"
+        },
+        {
+          "name": "senderAddr"
+        },
+        {
+          "name": "senderAddrLen"
+        },
+        {
+          "name": "senderPort"
+        },
+        {
+          "name": "senderScopeId"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.NativeStaticallyReferencedJniMethods",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.Buffer",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.DatagramSocketAddress",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "byte[]",
+            "int",
+            "int",
+            "int",
+            "io.grpc.netty.shaded.io.netty.channel.unix.DatagramSocketAddress"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.DomainDatagramSocketAddress",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "byte[]",
+            "int",
+            "io.grpc.netty.shaded.io.netty.channel.unix.DomainDatagramSocketAddress"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.ErrorsStaticallyReferencedJniMethods",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.FileDescriptor",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.LimitsStaticallyReferencedJniMethods",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.PeerCredentials",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "int",
+            "int",
+            "int[]"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.channel.unix.Socket",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.util.AbstractReferenceCounted",
+      "fields": [
+        {
+          "name": "refCnt"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.util.DefaultAttributeMap"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.netty.shaded.io.netty.util.internal.NativeLibraryUtil",
+      "methods": [
+        {
+          "name": "loadLibrary",
+          "parameterTypes": [
+            "java.lang.String",
+            "boolean"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "io.grpc.override.ContextStorageOverride"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.io.FileDescriptor",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "fd"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.io.IOException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.lang.BaseVirtualThread"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.lang.OutOfMemoryError",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.lang.RuntimeException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.lang.Thread",
+      "methods": [
+        {
+          "name": "isVirtual",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.net.InetSocketAddress",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "java.lang.String",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.net.PortUnreachableException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.Bits",
+      "fields": [
+        {
+          "name": "UNALIGNED"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.Buffer",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "address"
+        },
+        {
+          "name": "limit"
+        },
+        {
+          "name": "position"
+        }
+      ],
+      "methods": [
+        {
+          "name": "limit",
+          "parameterTypes": []
+        },
+        {
+          "name": "position",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "java.nio.Buffer",
+      "fields": [
+        {
+          "name": "address"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.ByteBuffer"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.DirectByteBuffer"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.channels.ClosedChannelException",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.nio.channels.FileChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "java.time.Instant"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "jdk.internal.misc.Unsafe",
+      "methods": [
+        {
+          "name": "getUnsafe",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "libcore.io.Memory"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "org.robolectric.Robolectric"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "sun.misc.Unsafe",
+      "fields": [
+        {
+          "name": "theUnsafe"
+        }
+      ],
+      "methods": [
+        {
+          "name": "invokeCleaner",
+          "parameterTypes": [
+            "java.nio.ByteBuffer"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.proto.ApiConfig"
+      },
+      "type": "sun.misc.Unsafe",
+      "fields": [
+        {
+          "name": "theUnsafe"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "type": "sun.nio.ch.FileChannelImpl",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "fd"
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "glob": "META-INF/native/libio_grpc_netty_shaded_netty_transport_native_epoll_x86_64.so"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "glob": "META-INF/services/io.grpc.ManagedChannelProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpMultiEndpointChannel"
+      },
+      "glob": "META-INF/services/io.grpc.NameResolverProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "glob": "META-INF/services/java.time.zone.ZoneRulesProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "module": "java.logging",
+      "glob": "sun/util/logging/resources/logging_en.properties"
+    },
+    {
+      "condition": {
+        "typeReached": "com.google.cloud.grpc.GcpManagedChannel"
+      },
+      "module": "java.logging",
+      "glob": "sun/util/logging/resources/logging_en_US.properties"
+    },
+    {
+      "bundle": "sun.util.logging.resources.logging"
+    }
+  ]
+}

--- a/metadata/com.google.cloud/grpc-gcp/index.json
+++ b/metadata/com.google.cloud/grpc-gcp/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "1.11.0",
+    "test-version" : "1.10.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/com/google/cloud/grpc-gcp/$version$/grpc-gcp-$version$-sources.jar",
+    "repository-url" : "https://github.com/googleapis/google-cloud-java",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/com/google/cloud/grpc-gcp/$version$/grpc-gcp-$version$-tests.jar",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/com/google/cloud/grpc-gcp/$version$/grpc-gcp-$version$-javadoc.jar",
+    "description" : "grpc-gcp provides Java extensions for using gRPC clients with Google Cloud Platform services. It adds a managed-channel layer with channel pooling, affinity configuration, multi-endpoint support, fallback handling, and metrics integration for Cloud API calls.",
+    "tested-versions" : [
+      "1.11.0"
+    ],
+    "allowed-packages" : [
+      "com.google.cloud.grpc"
+    ]
+  },
+  {
     "metadata-version" : "1.10.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/com/google/cloud/grpc-gcp/$version$/grpc-gcp-$version$-sources.jar",
     "repository-url" : "https://github.com/GoogleCloudPlatform/grpc-gcp-java",

--- a/stats/com.google.cloud/grpc-gcp/1.11.0/stats.json
+++ b/stats/com.google.cloud/grpc-gcp/1.11.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "1.11.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 6591,
+        "missed" : 9435,
+        "ratio" : 0.411269,
+        "total" : 16026
+      },
+      "line" : {
+        "covered" : 1760,
+        "missed" : 2201,
+        "ratio" : 0.444332,
+        "total" : 3961
+      },
+      "method" : {
+        "covered" : 409,
+        "missed" : 376,
+        "ratio" : 0.521019,
+        "total" : 785
+      }
+    }
+  } ]
+}

